From ebefa2bc4fd9dca4e61b4ae0998712d0e420c56b Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
Date: Tue, 18 Jul 2017 10:50:57 +0300
Subject: [PATCH] rosidl_typesupport_cpp: remove redundant target linking

It's enough that CMakeLists.txt contains

link_directories(${Poco_LIBRARY_DIR})

The line

target_link_libraries(${PROJECT_NAME} ${Poco_LIBRARIES})

also adds Poco's include dir (which is /usr/include) as a system one
and that leads to triggering the bug
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70129 in gcc >= v6.

| /home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot-native/usr/bin/x86_64-refkit-linux/x86_64-refkit-linux-g++   -m64 -march=corei7 -mtune=corei7 -mfpmath=sse -msse4.2 --sysroot=/home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot  -DPOCO_NO_AUTOMATIC_LIBS -DROSIDL_TYPESUPPORT_CPP_USE_POCO -Drosidl_typesupport_cpp_EXPORTS -I/home/rojkov/work/iot-ref-kit/build/workspace/sources/rosidl-typesupport-cpp/rosidl_typesupport_cpp/include -isystem /home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot/usr/include  -O2 -pipe -g -feliminate-unused-debug-types -fdebug-prefix-map=/home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0=/usr/src/debug/rosidl-typesupport-cpp/git-r0 -fdebug-prefix-map=/home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot-native= -fdebug-prefix-map=/home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot=  -fstack-protector-strong -pie -fpie -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -fvisibility-inlines-hidden -fPIC   -Wall -Wextra -Wpedantic -std=gnu++14 -o CMakeFiles/rosidl_typesupport_cpp.dir/src/type_support_dispatch.cpp.o -c /home/rojkov/work/iot-ref-kit/build/workspace/sources/rosidl-typesupport-cpp/rosidl_typesupport_cpp/src/type_support_dispatch.cpp
| In file included from /home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot/usr/include/c++/7.1.0/ext/string_conversions.h:41:0,
|                  from /home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot/usr/include/c++/7.1.0/bits/basic_string.h:6159,
|                  from /home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot/usr/include/c++/7.1.0/string:52,
|                  from /home/rojkov/work/iot-ref-kit/build/workspace/sources/rosidl-typesupport-cpp/rosidl_typesupport_cpp/src/type_support_dispatch.hpp:23,
|                  from /home/rojkov/work/iot-ref-kit/build/workspace/sources/rosidl-typesupport-cpp/rosidl_typesupport_cpp/src/service_type_support_dispatch.cpp:17:
| /home/rojkov/work/iot-ref-kit/build/tmp-glibc/work/corei7-64-refkit-linux/rosidl-typesupport-cpp/git-r0/recipe-sysroot/usr/include/c++/7.1.0/cstdlib:75:15: fatal error: stdlib.h: No such file or directory
|  #include_next <stdlib.h>
|                ^~~~~~~~~~
| compilation terminated.

Upstream-Status: Inappropriate [isn't reproducible in upstream official builds]
---
 rosidl_typesupport_cpp/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/rosidl_typesupport_cpp/CMakeLists.txt b/rosidl_typesupport_cpp/CMakeLists.txt
index f0ab0b8..7012311 100644
--- a/rosidl_typesupport_cpp/CMakeLists.txt
+++ b/rosidl_typesupport_cpp/CMakeLists.txt
@@ -43,7 +43,6 @@ target_include_directories(${PROJECT_NAME}
   PUBLIC
   include ${Poco_INCLUDE_DIRS}
 )
-target_link_libraries(${PROJECT_NAME} ${Poco_LIBRARIES})
 ament_target_dependencies(${PROJECT_NAME} "rosidl_generator_c")
 ament_export_libraries(${PROJECT_NAME})
 
-- 
2.9.3

